### DISCUSSION TITLE
how to set process.env for web3.js

### DISCUSSION THREAD

**Discussion by baixiaoyu2997** [nuxt/nuxt#17710] [Questions] [answered]
I didn't find any related questions in the discussion forum

im use nuxt3 ,when i use  `web3.js`  show me the error `Uncaught TypeError: Cannot read properties of undefined (reading 'NODE_DEBUG')`

i found web3 use `process.env.NODE_DEBUG`，but `vite` can't edit `process`
```js
if (process.env.NODE_DEBUG) {
  var debugEnv = process.env.NODE_DEBUG;
  debugEnv = debugEnv.replace(/[|\\{}()[\]^$+?.]/g, '\\$&')
    .replace(/\*/g, '.*')
    .replace(/,/g, '$|^')
    .toUpperCase();
  debugEnvRegex = new RegExp('^' + debugEnv + '$', 'i');
}
```
so, what should I do

**Comment by baixiaoyu2997** [comment, marked as answer]
https://github.com/nuxt/framework/discussions/4393

**Reply by Nookee411** [reply]
Link is invalid

**Comment by Nookee411** [comment]
If anyone stumble apon this issue, here's new link that helped me
https://github.com/browserify/node-util/issues/43#issuecomment-1046110526
